src/lib/git.sh

```diff
@@ -0,0 +1,51 @@
+# The git.sh file provides git easy functionalities for kworkflow code.
```

Comment:
Just one overall comment for this file. Keep in mind that the bar for lib files should be as high as possible since many other files will use it, and it is easy to add tests. Remember, a library is only useful if it is reliable; try to be very rigorous about validations and corner cases here.